### PR TITLE
feat: [FX-4053] create codemods for RTE

### PR DESCRIPTION
**Pull request by dmaklygin**
[FX-4053]

### Description

Create codemods for RTE migration to a separate package.

### How to test

- Use alpha version of `@toptal/picasso-codemod`

RUN 
```sh
npx @toptal/picasso-codemod@5.4.2-alpha-feat-rte-codemod-8a53efa8c.8+8a53efa8c v36.0.0 --run-in-band
```

Checked in Staff Portal: https://github.com/toptal/staff-portal/pull/10570

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Self reviewed
- [x] Covered with tests

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-4053]: https://toptal-core.atlassian.net/browse/FX-4053?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by dmaklygin** [comment]
@toptal-bot run package:alpha-release

**Comment by dmaklygin** [comment]
@toptal-anvil ping reviewers

**Comment by dmaklygin** [comment]
@toptal-anvil ping reviewers

File: packages/picasso-codemod/bin/cli.mjs [hunk 1]
```diff
@@ -28,7 +28,7 @@ const findFilesInMonorepo = () =>
   execaSync('find', [
     '.',
     '-name',
-    '*.tsx',
+    '*.ts*',
```
Comment (dmaklygin) [inline packages/picasso-codemod/bin/cli.mjs:31]: it used to take only TSX files

File: packages/picasso-codemod/bin/cli.mjs [hunk 2]
```diff
@@ -105,6 +106,7 @@ export const run = () => {
       
       Options
         --parser-config  Add parser config
+        --run-in-band    Run serially in the current process (default: false)
```
Comment (dmaklygin) [inline packages/picasso-codemod/bin/cli.mjs:109]: we have to execute this `codemod` sequentially

File: packages/picasso-codemod/src/v36.0.0/__testfixtures__/imports.output.tsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { RichTextEditor } from '@toptal/picasso-forms'
+import type { CustomEmojiGroup } from '@toptal/picasso-rich-text-editor/RichTextEditor';
+import type { CustomEmoji } from '@toptal/picasso-rich-text-editor/RichTextEditor/types';
+import type { RichTextEditorProps as LocalRichTextEditorProps } from '@toptal/picasso-rich-text-editor';
+import { Container, Typography } from '@toptal/picasso';
+import { RichText as LocalRichText } from '@toptal/picasso-rich-text-editor';
+import { RichText } from '@toptal/picasso-rich-text-editor';
+import { Icon } from '@toptal/picasso'
+import { Something, transformString } from 'some-module'
+import { htmlToHast } from '@toptal/picasso-rich-text-editor/utils';
```
Comment (OleksandrNechai) [inline packages/picasso-codemod/src/v36.0.0/__testfixtures__/imports.output.tsx:10]: Are we moving utils to `picasso-rich-text-editor`?

File: packages/picasso-codemod/src/v36.0.0/__testfixtures__/imports.output.tsx [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { RichTextEditor } from '@toptal/picasso-forms'
+import type { CustomEmojiGroup } from '@toptal/picasso-rich-text-editor/RichTextEditor';
+import type { CustomEmoji } from '@toptal/picasso-rich-text-editor/RichTextEditor/types';
+import type { RichTextEditorProps as LocalRichTextEditorProps } from '@toptal/picasso-rich-text-editor';
+import { Container, Typography } from '@toptal/picasso';
+import { RichText as LocalRichText } from '@toptal/picasso-rich-text-editor';
+import { RichText } from '@toptal/picasso-rich-text-editor';
```
Comment (OleksandrNechai) [inline packages/picasso-codemod/src/v36.0.0/__testfixtures__/imports.output.tsx:7]: Not related, but is not it better to name package `picasso-rich-text`? It is more generic and appropriate for package name than picasso-rich-text-editor IMO

File: packages/picasso-codemod/src/v36.0.0/rich-text-editor-replacement.ts [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+  // first of all, we need to search for a package json, it must be in the root of the package
+  updatePackageJsonVersions(file.path, {
+    addRichTextEditorDependency:
+      fileContainsRichTextEditorImport || hasPicassoForms,
+  })
+
+  return source.toSource({ trailingComma: false, quote: 'single' })
+}
+
+export default transform
```
Comment (OleksandrNechai) [inline packages/picasso-codemod/src/v36.0.0/rich-text-editor-replacement.ts:194]: GPT generated code? 😄 
Looks clean, I don't think it worth going deep trying to understand it 😄

**Review by OleksandrNechai — APPROVED** [review]
Looks good 👍

File: packages/picasso-codemod/src/v36.0.0/__testfixtures__/imports.output.tsx (hunk 3 above)
Comment (dmaklygin) [reply packages/picasso-codemod/src/v36.0.0/__testfixtures__/imports.output.tsx:10]: yes, it comes from the `RTE`

File: packages/picasso-codemod/src/v36.0.0/rich-text-editor-replacement.ts (hunk 5 above)
Comment (dmaklygin) [reply packages/picasso-codemod/src/v36.0.0/rich-text-editor-replacement.ts:194]: Partially generated :)

File: packages/picasso-codemod/src/v36.0.0/__testfixtures__/imports.output.tsx (hunk 4 above)
Comment (dmaklygin) [reply packages/picasso-codemod/src/v36.0.0/__testfixtures__/imports.output.tsx:7]: We've already created such package. Renaming it to the `picasso-rich-text` may take some time.
Do you think, it's a mandatory?

**Review by TomasSlama — APPROVED** [review]

File: packages/picasso-codemod/src/v36.0.0/__testfixtures__/imports.output.tsx (hunk 4 above)
Comment (OleksandrNechai) [reply packages/picasso-codemod/src/v36.0.0/__testfixtures__/imports.output.tsx:7]: no, NIT, but take it as a learning point

File: packages/picasso-codemod/src/v36.0.0/__testfixtures__/imports.input.tsx [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import { RichTextEditor } from '@toptal/picasso-forms'
+import type { CustomEmojiGroup } from '@toptal/picasso/RichTextEditor'
+import type { CustomEmoji } from '@toptal/picasso/RichTextEditor/types'
+import type { RichTextEditorProps as LocalRichTextEditorProps } from '@toptal/picasso'
+import {
+  Container,
+  RichText as LocalRichText,
+  Typography,
+} from '@toptal/picasso'
+import { RichText } from '@toptal/picasso'
```
[the remark this answers is not in the document]
Comment (dmaklygin) [reply packages/picasso-codemod/src/v36.0.0/__testfixtures__/imports.input.tsx:10]: Actually, we have `RichTextEditor` from the `picasso-forms`. I think, having such tests is enough

**Pull request merged**